src/Microsoft.SqlTools.ServiceLayer/LanguageServices/LanguageService.cs

```diff
@@ -543,6 +543,12 @@ private static TelemetryProperties CreatePeekTelemetryProps(bool succeeded, bool
                     GetSignatureHelp(textDocumentPosition, scriptFile)
                         .ContinueWith(async task =>
                     {
+                        if (task.IsFaulted)
```

Comment:
Are you able to debug into this?
I think faulted tasks continue to "ContinueWithOnFaulted" continuation.

Reply:
Yes, and I verified it works now (see the logs in the description).

It will go to ContinueWithOnFaulted if there is one - but since this doesn't then it will default to going into ContinueWith. So yes, we could add a separate ContinueWithOnFaulted - but I didn't see much benefit in doing that vs just manually checking IsFaulted

Reply:
That's strange.. but thanks for capturing it.

Reply:
More context - ContinueWithOnFaulted is our code wrote - and it's just using ContinueWith and IsFaulted anyways

https://github.com/microsoft/sqltoolsservice/blob/main/src/Microsoft.SqlTools.ServiceLayer/Utility/TaskExtensions.cs#L65

There's some additional stuff it doesn't, but I have a couple issues with it : 

1. I'm not sure why Exception would ever be non-null is IsFaulted is false...that seems redundant to me (and no examples I've seen do the same thing)
2. ContinueWithOnFaulted uses Wait() when calling the handler

One thing to keep in mind - ContinueWith isn't even called if an exception is thrown while starting the task. e.g. the situation I ran in to was that I had manually binplaced a file so there was a runtime exception trying to load a method that didn't exist. So when it tried to call GetSemanticMarkers it thew the runtime exception before the task actually started and thus ContinueWith never got called at all. That's why I had to add the try/catch. This is very much an edge case though so I'm not too worried about it - and generally most of our handler logic will capture any exceptions like that and send back an error immediately.